encord_agents/core/settings.py

```diff
@@ -31,14 +32,24 @@ class Settings(BaseSettings):
 
     @field_validator("ssh_key_file")
     @classmethod
-    def check_path_expand_and_exists(cls, path: Path | None):
+    def check_path_expand_and_exists(cls, path: Path | None) -> Path | None:
         if path is None:
-            return path
+            return None
 
         path = path.expanduser()
         assert path.is_file(), f"Provided ssh key file (ENCORD_SSH_KEY_FILE: '{path}') does not exist"
         return path
 
+    @field_validator("ssh_key_content")
+    @classmethod
+    def check_is_not_path(cls, ssh_key_content: str | None) -> str | None:
+        if ssh_key_content is None:
+            return None
+        if os.path.exists(ssh_key_content):
```

Comment:
Could try / except pathlib conversion to path but this doesn't raise exceptions and directly checks whether the value points to a file. (Could point to folder conceptually 🤔 )

Reply:
I think that I'd rather raise a more understandable error here instead of just magically solving the issue. Reason being that we're inheriting the env variables from the main SDK so wouldn't want to mess with that IMO.
So perhaps just 
```suggestion
        if os.path.exists(ssh_key_content):
            raise ValueError("It looks like your environment variable `ENCORD_SSH_KEY` is a file path and not the private ssh key content. Did you mean to set `ENCORD_SSH_KEY_PATH` instead?")
```

Reply:
I disagree. I don't see why we wouldn't just magical resolve the issue? The point is not being right, it's making it work appropriately. If customers have their env keys in one place, then they may rely on that behaviour.